Source/WebInspectorUI/UserInterface/Views/TimelineRecordingContentView.css

```diff
@@ -23,28 +23,30 @@
  * THE POSSIBILITY OF SUCH DAMAGE.
  */
 
+.content-view.timeline-recording {
+    display: flex;
+    flex-direction: column;
+    overflow: auto;
+
+    --timeline-content-browser-min-height: 250px;
+}
+
 .content-view.timeline-recording > .timeline-overview {
-    position: absolute;
-    top: 0;
-    left: 0;
-    right: 0;
+    position: relative;
+    flex: none;
 }
 
 .content-view.timeline-recording.edit-instruments > .timeline-overview {
-    bottom: 0;
+    flex: 1;
 }
 
 .content-view.timeline-recording.edit-instruments > .content-browser {
     display: none;
 }
 
 .content-view.timeline-recording > .content-browser {
-    position: absolute;
-    left: 0;
-    right: 0;
-    bottom: 0;
-    overflow: hidden;
-
+    flex: 1;
+    min-height: var(--timeline-content-browser-min-height);
```

Comment:
if the content of the selected timeline (or the overview) is itself sub-scrollable (e.g. via a `WI.DataGrid`) then i think we could get into a slightly weird situation where you cant easily scroll up unless you mouse over a small area of the UI (e.g. the `WI.NavigationBar` above the content) or have already scrolled all the way to the top of the sub-scrollable content.  has there been any discussion around our concern for this situation and/or possible mitigations?

Reply:
> ... you cant easily scroll up unless you mouse over a small area of the UI (e.g. the WI.NavigationBar above the content) or have already scrolled all the way to the top of the sub-scrollable content.

Yes, that can happen. But there is a workaround. With the default overscroll behaviour,  the scroll gesture from the nested content view can to pass up into the parent content view. So it is possible to scroll back up to the top of the Timelines tab if you first scroll up to the top of the `WI.DataGrid`. Not ideal, but not a scroll trap either. 

See screen recording:
https://github.com/WebKit/WebKit/assets/105809798/6429d455-cc1f-43f3-a6bc-c4d1d8059d89

One way to mitigate the double scroll containers: on short viewports, expand the nested content view to its full height (i.e. as tall as the `WI.DataGrid`), and have just one very tall scroll container for the entire Timelines tab.

The tricky part is that a lot of content inside the content view is absolute positioned, so now it does not contribute to its height: 

```css
.timeline-view.media > .data-grid {
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
}
```

Reply:
yeah my concern is precisely that you have to scroll to the top of the `WI.DataGrid` in order to get back to the overview

i doubt we'd want to have the `WI.DataGrid` expand to its full height as that might wreck any sort of virtualization we have (or could have) in order to not have a bazillion nodes in the DOM at the same time

perhaps we could just tweak the UI such that the height doesnt take up more than some percentage (e.g. 2/3) of the main content area so that at least some part of the `.timeline-overview` is always visible?